packages/openapi/src/decorators.ts

```diff
@@ -58,16 +57,16 @@ export const $extension: ExtensionDecorator = (
   extensionName: string,
   value: unknown,
 ) => {
-  let data = value;
   if (value && isType(value as any)) {
```

Comment:
should we not  just remove? dont think we do that check in other decorators

Reply:
I initially did that - if someone does pass a `Type` to `$extension` it causes TypeSpec to crash (believe due to open api3 trying to recursively render the Type and hitting stack size limits). Only took extra care since we know at least 1 case of someone trying to call `$extension` directly but I don't feel too strongly and could remove the extra diagnostic logic.

Reply:
can you do a compiler Assert then so it crash but doesn't carry a diagnostic?